language_service/src/protocol.rs

```diff
@@ -83,25 +77,23 @@ impl Eq for CompletionItem {}
 use std::hash::{Hash, Hasher};
 
 impl Hash for CompletionItem {
-    // exclude sort text for hashing
+    // use only label and kind for hashing
     fn hash<H: Hasher>(&self, state: &mut H) {
         self.label.hash(state);
         self.kind.hash(state);
-        self.detail.hash(state);
```

Comment:
I think we still need to hash on everything except sort text to be able to distinguish between two items that have the same label, but are actually different in the changes they will make.

Reply:
Discussed offline: there is a risk that if two signatures defined in different namespaces are identical, this hash will merge them together despite the fact that they would have resulted in separate auto-open behavior covered by the `addition_text_edits`. We decided that this should be handled in a separate PR.